R/g.plot5.R

```diff
@@ -198,39 +199,94 @@ g.plot5 = function(metadatadir=c(),dofirstpage=FALSE, viewingwindow = 1,f0=c(),f
         # Next pages with day specific graphs
         #get variables - activity:
         ACC = as.numeric(as.matrix(M$metashort[,metric])) * 1000
+        nonwearscore = as.numeric(as.matrix(M$metalong[,"nonwearscore"]))
+        
         time =  as.character(M$metashort[,1])
+        nw_time = as.character(M$metalong[,1])
         if (length(unlist(strsplit(time[1],"T"))) > 1) { # ISO timestamp format
           time = as.character(iso8601chartime2POSIX(time,desiredtz))
+          nw_time = as.character(iso8601chartime2POSIX(nw_time,desiredtz))
+        }
+        
+        sec = unclass(as.POSIXlt(time))$sec
+        min = unclass(as.POSIXlt(time))$min
+        hour = unclass(as.POSIXlt(time))$hour
+        
+        # Prepare nonwear information for plotting
+        NONWEAR = rep(NA,length(ACC))
+        day = unclass(as.POSIXlt(time))$mday
+        month = unclass(as.POSIXlt(time))$mon + 1
+        year = unclass(as.POSIXlt(time))$year + 1900
+        # take instances where nonwear was detected (on ws2 time vector) and map results onto a ws3 length vector for plotting purposes
+        if (sum(which(nonwearscore > 1))) {
+          nonwear_elements = which(nonwearscore > 1)
+          for (j in 1:length(nonwear_elements)) {
+            # could add try/catch in here in case 'which' fails..
+            match_loc = which(nw_time[nonwear_elements[j]]==time)
+            NONWEAR[match_loc:(match_loc+(ws2/ws3)-1)] <- 1
+          }
         }
+
+        INACT = rep(NA,length(ACC))  # PA vectors for plotting
```

Comment:
would be neater to code this up as:
`INACT = LIGPA = MODPA = VIGPA = rep(NA,length(ACC)) # PA vectors for plotting`

Reply:
good point